pkg/api/dump.go

```diff
@@ -129,14 +131,13 @@ func (s *service) prepareDump(ctx context.Context, state *task.ProcessState, arg
 	if err != nil {
 		return "", err
 	}
-
 	elapsed := time.Since(start)
 	stats.PrepareDuration = elapsed.Milliseconds()
 
 	return dumpDirPath, nil
 }
 
-func (s *service) postDump(ctx context.Context, dumpdir string, state *task.ProcessState, stream bool) {
+func (s *service) postDump(ctx context.Context, dumpdir string, state *task.ProcessState, stream bool, cmd *exec.Cmd) {
```

Comment:
I think you can get rid of an argument here and not pass the stream bool, and maybe rename cmd to streamCmd? Then just check if streamCmd is nil